ddtrace/tracer/remote_config.go

```diff
@@ -18,6 +20,38 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
 )
 
+type dynamicInstrumentationRCProbeConfig struct {
+	runtimeID     string
+	configPath    string
+	configContent string
+}
+
+type dynamicInstrumentationRCState struct {
+	mu    *sync.Mutex
```

Comment:
```suggestion
	sync.Mutex
```

To reduce clutter, embedding the mutex allows to call `{,R}{,Un}Lock` without writing `.mu.` and avoids using a pointer. Unless we plan to pass a variable of this type around by value, we don't need the pointer.

Reply:
Good call :+1: